tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsManager.cs

```diff
@@ -53,7 +53,16 @@ internal class DataStreamsManager
         IDiscoveryService discoveryService,
         string defaultServiceName)
     {
-        var writer = settings.IsDataStreamsMonitoringEnabled
+        var enableDsm = settings.IsDataStreamsMonitoringEnabled;
+        if (enableDsm && !settings.KafkaCreateConsumerScopeEnabled)
```

Comment:
I don't think we should, but still asking the question. 
Should we disable DSM if the Kafka Integration is disabled? It's not a big deal as it won't create incorrect results as here, but it's a few bytes of memory saved. As it must be a super rare scenario I don't think it's worth it but still asking :p

Reply:
Discussed offline, on the one hand, it makes sense. But as we expect to support other integrations in the future, I think it _maybe_ is more of a risk that we forget to remove that filter. WE also don't expect customers to enable DSM if they have disabled the kafka integration